test/e2e/capi_test.go

```diff
@@ -226,15 +226,30 @@ var _ = Describe("Running the Cluster API E2E tests", func() {
 				})
 				capi_e2e.ClusterctlUpgradeSpec(ctx, func() capi_e2e.ClusterctlUpgradeSpecInput {
 					return capi_e2e.ClusterctlUpgradeSpecInput{
-						E2EConfig:             e2eConfig,
-						ClusterctlConfigPath:  clusterctlConfigPath,
-						BootstrapClusterProxy: bootstrapClusterProxy,
-						ArtifactFolder:        artifactFolder,
-						SkipCleanup:           skipCleanup,
-						PreInit:               getPreInitFunc(ctx),
+						E2EConfig:                 e2eConfig,
+						ClusterctlConfigPath:      clusterctlConfigPath,
+						BootstrapClusterProxy:     bootstrapClusterProxy,
+						ArtifactFolder:            artifactFolder,
+						SkipCleanup:               skipCleanup,
+						PreInit:                   getPreInitFunc(ctx),
+						InitWithProvidersContract: "v1beta1",
 						ControlPlaneWaiters: clusterctl.ControlPlaneWaiters{
 							WaitForControlPlaneInitialized: EnsureControlPlaneInitialized,
 						},
+						InitWithBinary: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.0.5/clusterctl-{OS}-{ARCH}",
```

Comment:
So we have to maintain this going ahead? 
Is there a way we can reference the version number from environment variable?

Reply:
This will not have to be maintained as this test will always pull from version `v1.0.5`. We are planning to add more upgrade test from later versions much like how CAPI does here: https://github.com/kubernetes-sigs/cluster-api/blob/main/test/e2e/clusterctl_upgrade_test.go#L129-L130